src/haswell/bitmanipulation.h

```diff
@@ -36,14 +36,16 @@ really_inline int leading_zeroes(uint64_t input_num) {
   return static_cast<int>(_lzcnt_u64(input_num));
 }
 
-really_inline int count_ones(uint64_t input_num) {
 #ifdef _MSC_VER
+really_inline uint64_t count_ones(uint64_t input_num) {
   // note: we do not support legacy 32-bit Windows
   return __popcnt64(input_num);// Visual Studio wants two underscores
+}
 #else
+really_inline int count_ones(uint64_t input_num) {
```

Comment:
In my tests, this change here (return int) accounts for 50% of the regression...
```
setting it back to uint64_t: 2.4573 per byte
this PR: 2.4614 per byte
master: 2.4525 per byte
```

It is also unclear why you made this change at all given that you did not make this change on westmere.

Now... why would that make *any* difference to the performance. Who knows?

Reply:
I've had trouble with leading_zeroes and similar functions if the function returns anything other than the actual result type of the intrinsic. Might be I changed a function I didn't need to for some reason, I'll look at it--thanks.

Reply:
maybe try to replace _BitScanReverse64 with __lzcnt64?